tls/s2n_cipher_suites.c

```diff
@@ -1039,7 +1039,9 @@ static int s2n_set_cipher_and_cert_as_server(struct s2n_connection *conn, uint8_
                 continue;
             }
 
-            if (!s2n_kex_supported_and_configure(match, conn)) {
+            /* Short circuit logic to ensure the kex is supported (returns non zero value) and that the kex is
```

Comment:
This is a sufficiently complex description that I'd recommend breaking it into two separate if statements.